webapp/channels/src/components/widgets/advanced_textbox/advanced_textbox.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useState, useEffect} from 'react';
+
+import ShowFormat from 'components/advanced_text_editor/show_formatting/show_formatting';
+import Textbox from 'components/textbox';
+import type {TextboxElement} from 'components/textbox';
+import type TextboxClass from 'components/textbox/textbox';
+
+import './advanced_textbox.scss';
+
+type AdvancedTextboxProps = {
+    id: string;
+    value: string;
+    channelId: string;
+    onChange: (e: React.ChangeEvent<TextboxElement>) => void;
+    onKeypress: (e: React.KeyboardEvent<TextboxElement>) => void;
+    createMessage: string;
+    characterLimit: number;
+    minCharacterLimit?: number;
+    minLengthErrorMessage?: string;
+    preview: boolean;
+    togglePreview: () => void;
+    textboxRef?: React.RefObject<TextboxClass>;
```

Comment:
Could this be removed to avoid the new code using references to this? We're going to want to migrate that to be a functional component one of these days, so it would be good to prevent new code from using it without a good reason